docs/mine/lotus/dagstore.md

```diff
@@ -0,0 +1,191 @@
+2. Index repository (manages indices).
+3. DAG access layer (manages queries).
+
+## Terminology
+
+Here's a definition of terms:
+
+- **Shard:** the unit of data storage in the dagstore. Every Filecoin piece (the data within a deal) is a unique shard in the dagstore. Every shard is identified by a unique **key**, whose value is the **Piece CID**.
+- **Shard state:** the state that a dagstore shard is in. It can be one of the  following:
+  - `ShardStateNew`: indicates that a shard has been registered, but has not yet been initialized.
+  - `ShardStateInitializing`: indicates that the shard is being initialized.
+    - `ShardStateAvailable`: indicates that the shard has been initialized and is capable of serving retrievals. However, there are no active shard readers in this instant.
+  - `ShardStateServing`: indicates the shard has active readers, and thus is currently serving at least one retrieval.
+    - `ShardStateErrored`: indicates that an unexpected error was encountered during a shard operation (e.g. initialization or acquisition), and therefore the shard needs to be recovered to be fully operational.
+    - `ShardStateRecovering`: indicates that the shard is attempting to recover from an errored state.
+- **Shard initialization:** the act of fetching the shard's data, indexing it and storing the index under the `dagstore/index` directory.
+- **Shard acquisition:** the act of fetching the shard's data, joining it with an index, and building an accessor for shard data, in order to serve retrievals.
+- **Index:** a mapping of CID => CAR file offset, associated with a shard.
+- **Mount:** a key feature of the dagstore is location-transparency of shard data, and the mount is the component that teaches the dagstore how to obtain the data for a shard. A shard can be mounted from a local path, HTTP URL, distributed filesystem, the Lotus storage subsystem (precisely the mount type that we use in Lotus), or anything else.
+
+## Directory structure
+
+By default, the dagstore root will be placed in:
+
+- `$LOTUS_MARKETS_PATH/dagstore`, if you're running a [split miner/market deployment](./split-markets-miners.md).
+- `$LOTUS_MINER_PATH/dagstore`, if you're not.
+
+The directory structure is as follows:
+
+```
+ dagstore
+     |___ index                         # (1)
+     |___ transients                    # (2)
+     |___ datastore                     # (3)
+     |___ .shard-registration-complete  # (4)
+```
+
+1. `index`: holds the shard indices.
+2. `transients`: holds temporary shard data (unsealed pieces) while they're being indexed.
+3. `datastore`: records shard state and metadata so it can survive restarts.
+4. `.shard-registration-complete`: marker file that signals that initial migration is complete.
+
+## First-time migration
+
+When you first start 1.11.2, a migration process will register all shards in **lazy initialization** mode. As deals come in, shards are fetched and initialized just in time in order to serve the retrieval.
```

Comment:
Mention what miners should expect under this migration process - understand it’s gonna be a light process and has minimal impact on miner process - we should explicitly mention it

Reply:
Yes, feel free to add a clarifying comment.